### DISCUSSION TITLE
bitlang.cobol extension used by vscode-ibmi conflicts with that used by Z Open Editor

### DISCUSSION THREAD

**Discussion by jwoehr** [codefori/vscode-ibmi#1443] [Ideas]
### Problem

The bitlang.cobol extension used by vscode-ibmi conflicts with that used by Z Open Editor

![image](https://github.com/halcyon-tech/vscode-ibmi/assets/4604036/3b096c46-a0e6-4323-876d-ea9a32f9467d)

### Workaround

Disable Z Open Editor while working in vscode-ibmi

###  Desired Solution

Coordination between Code for i and the Z Open Editor project.

See [this Z Open Editor issue](https://github.com/IBM/zopeneditor-about/issues/352)
<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.0.0|
|Visual Studio Code version|1.80.1|
|Operating System|linux_x64|

<details>
<summary>Active extensions</summary>

```
COBOL (cobol): 9.7.15
Code Coverage for IBM i (code-coverage-ibmi): 0.1.7
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.3.1
CodeLLDB (vscode-lldb): 1.9.2
Db2 for IBM i (vscode-db2i): 0.2.11
Dev Containers (remote-containers): 0.299.0
ESLint (vscode-eslint): 2.4.2
EditorConfig for VS Code (EditorConfig): 0.16.4
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.12.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
GitHub Pull Requests and Issues (vscode-pull-request-github): 0.68.1
IBM Db2 for z/OS Developer Extension (db2forzosdeveloperextension): 2.1.1
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
JSON Language Features (json-language-features): 1.0.0
Live Share (vsliveshare): 1.0.5873
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
MongoDB for VS Code (mongodb-vscode): 1.1.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Path Intellisense (path-intellisense): 2.8.4
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
Vue Language Features (Volar) (volar): 1.8.5
Zowe Explorer (vscode-extension-for-zowe): 2.9.1
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|7|
|CCSID|37|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin||/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|
|-|-|-|-|-|
|bash|attr|ILEDITOR.GETNEWLIBL|QZDFMDB2.PGM|startDebugService.sh|
|chsh|iconv||||
|git|setccsid||||
|grep|||||
|ls|||||
|md5sum|||||
|sort|||||
|stat|||||
|tar|||||
|tn5250|||||
</details>

<details>
<summary>Shell env</summary>

```bash
deleted
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "system \"CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')\"",
    "code": 255,
    "stderr": "CPF2111: Library ILEDITOR already exists.",
    "cwd": "/home/jwoehr/builds/Manzan"
  }
]
```
</details>

**Comment by worksofliam** [comment]
Change to be a discussion instead of an issue. Sadly, there is nothing we can do here since both those extensions (Z Open Editor and COBOL) both provide `COBOL` as a language provider, among other things.

**Reply by jwoehr** [reply]
Just suggesting some co-ordination.
Classic IBM problem.
Maintaining the  i _vs_ z silos is outdated now that we're all in the same canoe and paddling towards modernization with the same paddles.

**Reply by worksofliam** [reply]
As a note: Code for IBM i does not have a hard requirement on bitlang.cobol. We are not bothered about what COBOL language tools the user users, but we do install it with the extension pack because it's free. As you discovered, you can disable it. This is something that Z Open Editor really needs to take of for now.

**Reply by jwoehr** [reply]
I did open an issue with them.
And I definitely do need COBOL in vscode-ibmi .. customer requirement.

**Comment by jwoehr** [comment]
Anyway, the Z folks suggested I use multiple profiles.
Which works.
I guess this is really a non-issue
